alica_engine/src/engine/AlicaContext.cpp

```diff
@@ -24,6 +24,9 @@ AlicaContext::AlicaContext(const AlicaContextParams& alicaContextParams)
         , _clock(std::make_unique<AlicaClock>())
         , _localAgentName(alicaContextParams.agentName)
 {
+    LockedBlackboardRW gbb(_globalBlackboard);
+    gbb.set("agentName", alicaContextParams.agentName);
+    gbb.set("agentID", alicaContextParams.agentID);
```

Comment:
The AlicaContextParam struct allows us to not pass an agentId and by default initializes it with an invalid ID. Because the config is already loaded from the initializer list you can check both the context and the config for having a correct ID. 

Are we syncing these ids later on? What happens if we have no ID set in the config file but pass one via the params? Will the teammanager still generate another ID? If the config has an ID that differs from the params, which one will be used by the agent?

Reply:
good catch. I fixed it by moving the setting logic to after engine->init()